src/service/genesis_gen.h

```diff
@@ -292,7 +292,8 @@ namespace ccf
       auto service = tx.rw(tables.service);
       service->put(
         {service_cert,
-         recovering ? ServiceStatus::RECOVERING : ServiceStatus::OPENING});
+         recovering ? ServiceStatus::RECOVERING : ServiceStatus::OPENING,
```

Comment:
I wasn't aware of the new `RECOVERING` status but I know some things (e.g. `ledger_viz.py`) rely on the `OPENING` status to detect a new service/recovery. I'd like to keep the recovery and service opening sequence as similar as possible because the two flows are already too complex (especially the latter). 

I can see the `RECOVERING` state only used to check that we require a `previous_service_identity` on the `transition_service_to_open` proposal. I feel like we could check this with the new `previous_service_identity_version` field in the service info table (i.e. if `std::nullopt`, then it's the first service open, otherwise it's a recovery).

Reply:
That's a good observation and I think you're right that the previous version should be enough of an indicator. On the side I'm always thinking about recovering a 2.x service from a 1.x ledger, and in that scenario, it wouldn't work, but maybe we just don't want to support that.

Reply:
Good point. I think it's OK to not support that for historical 1.x ledgers.

Reply:
Actually, halfway through making those changes, I realized that this is already in RC4. Is it still okay to remove this state, or do the same arguments as for the renaming of arguments apply?

Reply:
I'm leaving this in; that can be a separate PR if we decide to remove that state.

Reply:
Could you create an issue for this as this is important for some of our tooling?